### PR TITLE
Add new itest RecordStreamValidator using SignedStateHolder

### PR DESCRIPTION
**Pull request by MiroslavGatsanoga**
**Description**:

- Define new `AccountAliasesValidator.java` that will read all aliases from the transaction records and check that they match the ones in state. For reading the signed state file use the existing `com.hedera.services.cli.signedstate.SignedStateHolder` class
- In order to reuse the `SignedStateHolder` logic it seemed like I needed to define a `module-info.java` for `test-clients` and also update some of the module and build configuration

**Related issue(s)**:

Fixes https://github.com/hashgraph/hedera-services/issues/4649

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by david-bakin-sl — DISMISSED** [review]
LGTM.  Confirmed (on my machine) that the module-info changes are correct (and minimal). (I had to rebase from develop before it would compile for me.)

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/AccountAliasesValidator.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                    }
+                } catch (final InvalidProtocolBufferException e) {
+                    throw new IllegalStateException(e);
+                }
+            }
+        }
+    }
+
+    private void getActualAliasesFrom(final Path signedStateFile) throws Exception {
+        var state = new SignedStateHolder(signedStateFile).getPlatformState();
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/AccountAliasesValidator.java:100]: This won't work in general: The platform state is only valid while the `SignedStateHolder` exists.  A `SignedStateHolder` is an `AutoCloseable` so hold it in a try-with-resources.

**Comment by Nana-EC** [comment]
@MiroslavGatsanoga should this be picked up and resolved in 0.46 or is it blocked or no longer applicable?

**Comment by MiroslavGatsanoga** [comment]
> @MiroslavGatsanoga should this be picked up and resolved in 0.46 or is it blocked or no longer applicable?

@Nana-EC This draft was an initial attempt to add this new `AccountAliasesValidator` in mono but I am not sure if we want to do that yet and instead implement just it against modularized instead 🤔.

If that is the way we move forward with this I would say that a proper time to implement this is when we figure out how to run the other validators from mono in modularized (https://github.com/hashgraph/hedera-services/issues/8949).

I am not confident to say it will make it in 0.46 though, as there is still some work left in the modularization effort. cc @netopyr

**Comment by tinker-michaelj** [comment]
Stale (and these validators are not even running currently b/c of intermittent failures in CI), we can revisit later if desired.

**Pull request closed without merging**